uaclient/files/notices.py

```diff
@@ -0,0 +1,244 @@
+    )
+    FIPS_SYSTEM_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_system_reboot_required",
+        order_id="21",
+        is_permanent=False,
+        message=messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg,
+    )
+    FIPS_INSTALL_OUT_OF_DATE = NoticeFileDetails(
+        label="fips_install_out_of_date",
+        order_id="22",
+        is_permanent=True,
+        message=messages.FIPS_INSTALL_OUT_OF_DATE,
+    )
+    FIPS_DISABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_disable_reboot_required",
+        order_id="23",
+        is_permanent=False,
+        message=messages.FIPS_DISABLE_REBOOT_REQUIRED,
+    )
+    FIPS_PROC_FILE_ERROR = NoticeFileDetails(
+        label="fips_proc_file_error",
+        order_id="24",
+        is_permanent=True,
+        message=messages.FIPS_PROC_FILE_ERROR,
+    )
+    FIPS_MANUAL_DISABLE_URL = NoticeFileDetails(
+        label="fips_manual_disable_url",
+        order_id="25",
+        is_permanent=True,
+        message=messages.NOTICE_FIPS_MANUAL_DISABLE_URL,
+    )
+    WRONG_FIPS_METAPACKAGE_ON_CLOUD = NoticeFileDetails(
+        label="wrong_fips_metapackage_on_cloud",
+        order_id="25",
+        is_permanent=True,
+        message=messages.NOTICE_WRONG_FIPS_METAPACKAGE_ON_CLOUD,
+    )
+    LIVEPATCH_LTS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="lp_lts_reboot_required",
+        order_id="30",
+        is_permanent=False,
+        message=messages.LIVEPATCH_LTS_REBOOT_REQUIRED,
+    )
+    CONTRACT_REFRESH_WARNING = NoticeFileDetails(
+        label="contract_refresh_warning",
+        order_id="40",
+        is_permanent=True,
+        message=messages.NOTICE_REFRESH_CONTRACT_WARNING,
+    )
+    OPERATION_IN_PROGRESS = NoticeFileDetails(
+        label="operation_in_progress",
+        order_id="60",
+        is_permanent=True,
+        message="Operation in progress: {operation}",
+    )
+    AUTO_ATTACH_RETRY_FULL_NOTICE = NoticeFileDetails(
+        label="auto_attach_retry_full_notice",
+        order_id="70",
+        is_permanent=True,
+        message=messages.AUTO_ATTACH_RETRY_NOTICE,
+    )
+    AUTO_ATTACH_RETRY_TOTAL_FAILURE = NoticeFileDetails(
+        label="auto_attach_total_failure",
+        order_id="71",
+        is_permanent=True,
+        message=messages.AUTO_ATTACH_RETRY_TOTAL_FAILURE_NOTICE,
+    )
+
+
+class Notices:
+    def __init__(
+        self,
+    ):
+        pass
+
+    def add(
+        self,
+        root_mode: bool,
+        notice_details: NoticeFileDetails,
+        description: str,
+    ):
+        """Adds a notice file.
+
+        :param notice_details: Holds details concerning the notice file.
+        :param description: The content to be written to the notice file.
+        """
+        if root_mode:
+            label = notice_details.label
+            file_order_id = notice_details.order_id
+            is_notice_permanent = notice_details.is_permanent
+            self.directory = (
```

Comment:
This doesn't need to be an instance variable
```suggestion
            directory = (
```